### PR TITLE
Refresh README

### PR DESCRIPTION
**Pull request by tianchu**
### What does this PR do?

Refresh the README.

- Adjust parameters sequence by importance and relevance. Refreshed the out dated language.
- Remove bunch of notes regarding webpack, which has been covered in its own guide.
- Remove a note about forwarder subscription issue that has no longer been a problem

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

**Comment by alai97** [comment]
Created DOCS-3496!

File: README.md [hunk 1]
```diff
@@ -21,72 +13,70 @@ The plugin automatically enables instrumentation for applications to collect met
...
-| `customHandler`               | When set, the specified handler is set as the handler for all the functions. By default, the handler is set to `/opt/nodejs/node_modules/datadog-lambda-js/handler.handler` if `addLayers` is set to `true`, or `node_modules/datadog-lambda-js/dist/handler.handler` if `addLayers` is set to `false`.                                                                                                                             |
-| `failOnError`                 | When set, this plugin will throw an error if any custom Datadog monitors fail to create or update. This occurs after deploy, but will cause the result of `serverless deploy` to return a nonzero exit code (to fail user CI). Defaults to `false`.                                                                                                                                                                                 |
+| `site`                        | Set which Datadog site to send data to, such as `datadoghq.com` (default), `datadoghq.eu`, `us3.datadoghq.com`, `us5.datadoghq.com`, and `ddog-gov.com`. This parameter is required when collecting telemtry using the Datadog Lambda Extension. |
```
Comment (cswatt) [inline README.md:30]:
```suggestion
| `site`                        | Set which Datadog site to send data to, such as `datadoghq.com` (default), `datadoghq.eu`, `us3.datadoghq.com`, `us5.datadoghq.com`, or `ddog-gov.com`. This parameter is required when collecting telemtry using the Datadog Lambda Extension. |
```

File: README.md [hunk 2]
```diff
@@ -21,72 +13,70 @@ The plugin automatically enables instrumentation for applications to collect met
...
+| `apiKey`                      | [Datadog API key][7]. This parameter is required when collecting telemtry using the Datadog Lambda Extension. Alternatively you can set the `DATADOG_API_KEY` environment variable in your deployment environment. |
```
Comment (cswatt) [inline README.md:31]:
```suggestion
| `apiKey`                      | [Datadog API key][7]. This parameter is required when collecting telemetry using the Datadog Lambda Extension. Alternatively, you can also set the `DATADOG_API_KEY` environment variable in your deployment environment. |
```

File: README.md [hunk 3]
```diff
@@ -21,72 +13,70 @@ The plugin automatically enables instrumentation for applications to collect met
...
+| `appKey`                      | Datadog app key. Only needed when the `monitors` field is defined. Alternatively you can set the `DATADOG_APP_KEY` environment variable in your deployment environment. |
```
Comment (cswatt) [inline README.md:32]:
```suggestion
| `appKey`                      | Datadog app key. Only needed when the `monitors` field is defined. Alternatively, you can also set the `DATADOG_APP_KEY` environment variable in your deployment environment. |
```

File: README.md [hunk 4]
```diff
@@ -21,72 +13,70 @@ The plugin automatically enables instrumentation for applications to collect met
...
+| `apiKeySecretArn`             | An alternative to using the `apiKey` field. The ARN of the secret storing the Datadog API key in AWS Secrets Manager. Remember to add the `secretsmanager:GetSecretValue` permission to the Lambda execution role. |
```
Comment (cswatt) [inline README.md:33]:
```suggestion
| `apiKeySecretArn`             | An alternative to using the `apiKey` field. The ARN of the secret that is storing the Datadog API key in AWS Secrets Manager. Remember to add the `secretsmanager:GetSecretValue` permission to the Lambda execution role. |
```

File: README.md [hunk 5]
```diff
@@ -21,72 +13,70 @@ The plugin automatically enables instrumentation for applications to collect met
...
+| `env`                         | Tag your Datadog telemetry with the desired `env`. |
+| `service`                     | Tag your Datadog telemetry with the desired `service`. |
+| `version`                     | Tag your Datadog telemetry with the desired `version`. |
+| `enableXrayTracing`           | Set `true` to enable X-Ray tracing on the Lambda functions and API Gateway integrations. Defaults to `false`. |
+| `enableDDTracing`             | Enable Datadog tracing on the Lambda function. Defaults to `true`. |
+| `enableDDLogs`                | Enable Datadog log collection using the Lambda Extension. Defaults to `true`. Note: This setting has no effect on logs sent by the Datadog Forwarder. |
+| `monitors`                    | When defined, the Datadog plugin will configure monitors for the deployed function. Requires setting `DATADOG_API_KEY` and `DATADOG_APP_KEY` in your environment. To learn how to define monitors, see [To Enable and Configure a Recommended Serverless Monitor](#to-enable-and-configure-a-recommended-serverless-monitor). |
```
Comment (cswatt) [inline README.md:41]:
```suggestion
| `monitors`                    | When defined, the Datadog plugin configures monitors for the deployed function. Requires setting `DATADOG_API_KEY` and `DATADOG_APP_KEY` in your environment. To learn how to define monitors, see [To Enable and Configure a Recommended Serverless Monitor](#to-enable-and-configure-a-recommended-serverless-monitor). |
```

File: README.md [hunk 6]
```diff
@@ -21,72 +13,70 @@ The plugin automatically enables instrumentation for applications to collect met
...
+| `exclude`                     | When set, this plugin ignores all specified functions. Use this parameter if you have any functions that should not include Datadog functionality. Defaults to `[]`. |
+| `enabled`                     | When set to `false`, the Datadog plugin stays inactive. Defaults to `true`. You can control this option using an environment variable. For example, use `enabled: ${strToBool(${env:DD_PLUGIN_ENABLED, true})}` to activate/deactivate the plugin during deployment. Alternatively, you can also use the value passed in through `--stage` to control this option—[see example](#disable-plugin-for-particular-environment). |
+| `customHandler`               | When set, the specified handler is set as the handler for all the functions. |
+| `failOnError`                 | When set, this plugin will throw an error if any custom Datadog monitors fail to create or update. This occurs after deploy, but will cause the result of `serverless deploy` to return a nonzero exit code (to fail user CI). Defaults to `false`. |
```
Comment (cswatt) [inline README.md:53]:
```suggestion
| `failOnError`                 | When set, this plugin throws an error if any custom Datadog monitors fail to create or update. This occurs after deploy, but will cause the result of `serverless deploy` to return a nonzero exit code (to fail user CI). Defaults to `false`. |
```

File: README.md [hunk 7]
```diff
@@ -21,72 +13,70 @@ The plugin automatically enables instrumentation for applications to collect met
...
-    injectLogContext: true
     exclude:
       - dd-excluded-function
 ```
 
-**Note**: If you use webpack, Datadog recommends using the prebuilt layers by setting `addLayers` to `true`, which is the default, and add `datadog-lambda-js` and `dd-trace` to the [externals][10] section of your webpack config.
+### Webpack
 
-**Note**: Many of the above configuration values require setting `DATADOG_API_KEY` in your environment. If you store these values in AWS Secrets Manager or AWS KMS, refer to the above options `apiKeySecretArn` and `apiKMSKey` to pull your Datadog API key from the correct location.
+If you are using a bundler, such as webpack, see this [guide](https://docs.datadoghq.com/serverless/guide/serverless_tracing_and_webpack/).
```
Comment (cswatt) [inline README.md:74]:
```suggestion
If you are using a bundler, such as webpack, see [Serverless Tracing and Webpack](https://docs.datadoghq.com/serverless/guide/serverless_tracing_and_webpack/). 
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**